### ISSUE TITLE
turn the ship

### ISSUE THREAD

**Issue by whit537** [gratipay/gratipay.com#3399]
Last week, Gratipay [stopped processing](https://medium.com/gratipay-blog/gratipocalypse-42fd0ec0d9e8) payments because of legal concerns with our old terms of service that became apparent after our processor, Balanced, announced that they're going out of business, and Stripe wouldn't take us. So in addition to migrating our processing infrastructure (#3377), we also have to change our business model at the same time. No :sweat:, right? :)

We hashed out our new business model on https://github.com/gratipay/inside.gratipay.com/issues/180 and https://github.com/gratipay/inside.gratipay.com/issues/192. This ticket is to track implementation. See also the [Pivot](https://github.com/gratipay/gratipay.com/milestones/Pivot) milestone. 
##

**Comment by whit537** [comment]
I'm stuck thinking about the relationship between individuals, groups, and projects. I think #3337 is the cleanest way to go:

>Users optionally have Projects.
Projects optionally have Payroll.

Projects would be what you give money to.

But right now Users are what you give money to. We don't want to drop existing funding streams if we can at all help it. We want to give Users a way to migrate to a Project. Forcing migration before we process payments is okay since we're requiring acceptance of our new terms anyway.

**Comment by whit537** [comment]
I'm not sure how to pull of this pivot without #3337. That's a fair chunk of work.

**Comment by whit537** [comment]
Let's think about **Individuals** (formerly Participants; Users above) and **Teams** (vs. Projects above).

How do we migrate giving from Individuals to Teams? We basically need Individuals to create a new Team, to which we will reattach their giving.

A Team is a group of people working to deliver a given product or service.

Teams have Members. Some Members are Owners.

Since giving to Individuals has been ambiguous in regard to products or services delivered, migrating giving to a single-purpose Team will be a judgement call for our users. If I have 100 supporters, and 60 would say they were giving to me because of one product or service I provide, and 30 because of another, and 10 just because they love me, then how should I migrate? I don't even know who my supporters are, so it's not like I can manually review and make assignments ("Oh, this person is a Fizzbuzz user, that's probably why they've been supporting me. Let's move their payment to the new Fizzbuzz Team"). No: I have to guess at the 60-supporter product/service, and migrate all of my giving to a new Team dedicated to that. Then I can make a second Team for the 30-supporter product/service.

**Comment by whit537** [comment]
Can a Team give to another Team?

**Comment by whit537** [comment]
> Can a Team give to another Team?

I think not, but I think the Owner of a Team can give to other Teams. Let's say that Exchange Routes still attach to Participants. The Owner of a Team is a Participant, so we still have the chance to offset their giving with their receiving in a given cycle.

**Comment by whit537** [comment]
The approach I'm considering is to make new tables `teams`, `subscriptions`, `payroll`, `payments` that are parallel to `participants`, `tips`, `takes`, `transfers`. We no longer write to the old tables, but they're available to use for migrating. Migrating happens when a Participant takes the initiative; we're not migrating all `tips` at once.

**Comment by whit537** [comment]
Where do we put Teams in our URL structure? And in our [IA](http://inside.gratipay.com/big-picture/product), for that matter?

Do we demote Participants? https://gratipay.com/whit537 → https://gratipay.com/~whit537. Does that redirect? Almost certainly, or we'd break lots of inbound links.

But then does https://gratipay.com/~Gratipay redirect back to https://gratipay.com/Gratipay? Do we have a single namespace for Participant and Team slugs, even if we differentiate them in the URL? Probably. It doesn't seem right to allow someone to register https://gratipay.com/~Gratipay. On the other hand, we're saying that we want an Owner Participant for a Team, so I guess we *should* allow ~Gratipay to be the Owner of Gratipay. Right?

**Comment by whit537** [comment]
In terms of IA, I think we're looking at the following changes as the absolute minimum to pull this off this week:

 - `/$USERNAME/` → `/~$USERNAME/`
 - Add `$TEAM` at `/$TEAM/`, with a `Profile` at top level.
 - `/$USERNAME/members/` → `/$TEAM/payroll/`.

**Comment by whit537** [comment]
How do Participants register new Teams?

**Comment by whit537** [comment]
How about adding `/new` for that? Then we add `/dashboard/teams` for the new Teams review queue.

**Comment by whit537** [comment]
The first time a Participant with Tips creates a new Team, we should migrate all their Tips to Subscriptions for the Team.

**Comment by whit537** [comment]
This feels good. I like how this is coming together, it's an evolution of Teams, a pretty natural step.

**Comment by whit537** [comment]
We don't need pledging any longer, which means we can cut out almost all of the simplates under `/on/%platform/`.

**Comment by whit537** [comment]
I guess it's still useful to redirect from there to a connected Gratipay account.

**Comment by whit537** [comment]
- [ ] replace UI over `tips` & `takes` with UI over `subscriptions` & `payroll` (#3400)
 - [ ] implement UI for making/migrating to a new team
 - [ ] revise payday to use `subscriptions` `payroll` instead of `tips` `takes`

**Comment by whit537** [comment]
- [ ] unroll the interaction of `number` w/ `takes`
 - [ ] text fields on `teams` want to be i18n'd like statement

**Comment by whit537** [comment]
After bashing around on #3400 all day yesterday, I'm starting to get a better idea of how to approach this. I've made three clean PRs:

 - #3402 makes room in the URL structure for Teams
 - #3403 cuts out pledging
 - #3405 cuts out goals

We should be able to clean up tip limits based on number as well. Is there any other code we can simply cut out?

Beyond that, here's a review of work based on looking at a profile (**hard**, *medium*, easy):

 - [ ] Dashboard - Quarter-baked as it is. Copy is all wrong but nothing besides that.
 - [ ] *Profile* - Teams section will need to work with new Teams.
 - [ ] **Members** - This will be moved entirely to Teams. I'm not sure how `anonymous_receiving` interacts with this but that's probably out of scope for this ticket.
 - [ ] *Receiving* - Still want this, but it should only show payments from teams under the new system. It should also continue to respect `anonymous_receiving`.
 - [ ] *Giving* - Still want this, but it should only show payments to teams under the new system. It should also continue to respect `anonymous_giving`.
 - [ ] *History* - This should show history under both the old and new systems.
 - [ ] **Widgets** - Widgets are going to have to change somehow as part of this transition. How?
 - [ ] **Identity** - We almost certainly want identity information for any Team Owner. Right now we store this info at Balanced (they verify it with some API we don't know about). Do we need to start storing identity info locally as part of this ticket, even if we don't programmatically verify it yet?
 - [ ] Settings - I think this is pretty much fine. Notifications looks goofier than every without the "notify patrons" option, but I'm not going to worry about it here.
 - [x] Events - No references to other tables here, so I think we're good.

**Comment by whit537** [comment]
What should we do with `tips.json`? I think we should make a new API that uses team id instead, and just let tips.json go 404. How are we supposed to interpret tips to usernames from the old tipping system under the new system?

**Comment by whit537** [comment]
In which case, I think I'll rip out `tips.json` for this week, and reticket bringing back something similar.

**Comment by whit537** [comment]
**URL structure**

 1. [ ] move `/username/` to `/~username/` (#3402)

**Archive `tips`**

 1. [ ] remove pledging entirely (#3403)
 1. [ ] turn off giving (UI, JSON API, Python API)
 1. [ ] move `tips` to `old_tips`

**Archive `takes`**

 1. [ ] turn off taking (UI, JSON API, Python API)
 1. [ ] move `takes` to `old_takes`

**Clean up Participant**

 1. [ ] remove `goal` (#3405)
 1. [ ] remove varying limits based on `number`
 1. [ ] remove `receiving`
 1. [ ] remove `tips.json`

**`transfers`**

 1. [ ] rewrite History to use `old_transfers`
 1. [ ] move `transfers` to `old_transfers`

**Create Team**

 1. [ ] make new `tips`, `takes`, and `transfers` tables
 1. [ ] promote `_mixin_team` to a `Team` class
 1. [ ] implement team creation
 1. [ ] implement giving to teams
 1. [ ] implement taking from teams
 1. [ ] implement team migration
 1. [ ] implement team approval

**Rewrite payday*

**Comment by whit537** [comment]
I have the feeling that I'm going to be doing most of this work myself. I'm not even sure we'll be able to wait for proper review before merging and deploying. I'm at least going to keep using PRs, even if I end up merging them myself.

**Comment by whit537** [comment]
I went ahead and merged/deployed #3402, to get the ball rolling.

**Comment by whit537** [comment]
From #3406, here's a census of the parts of the user experience that touch the tips table:

 - closing accounts
 - updating team takes
 - adding/updating exchange routes
 - setting tips
 - merging accounts
 - tips.json
 - the old button widget
 - the pricing page
 - the stats page
 - the receiving page
 - the giving page
 - archiving accounts

Here's the details:

Here's what I can find:

```
✓ _check_tips
✓ _check_orphans_no_tips

✓ Participant.update_number
  Participant.clear_tips_giving - close
  Participant.clear_tips_receiving - close
  Participant.update_giving - team takes, exchange routes, setting tips, take over
  Participant.update_receiving - team takes, setting tips, take over
  Participant.set_tip_to - {tip,tips}.json, close, throughout the test suite
  Participant.get_tip_to - tip widget, old button widget, pricing
  Participant.get_tip_distribution - stats, receiving page
  Participant.get_giving_for_profile - giving page, tips.json, close
  Participant.get_tips_receiving - unused!
  Participant.get_current_tips - tips.json
  Participant.take_over - take over
  Participant.final_check - archive, close

  www/about/stats.spt
  www/about/tip-distribution.json.spt
```

**Comment by whit537** [comment]
I'm deploying #3403.

**Comment by whit537** [comment]
Looks like we have a regression from #3402: the tip widget is posting to `/foo/tip.json`, which is 302 to a GET and the POST body is lost. I'm going to proceed with #3406, which simply makes `tip.json` 404 ahead of the move to `/team/tip.json`.

**Comment by whit537** [comment]
Okay, I'm pumped. :boom: :exclamation: 

```python
#!/usr/bin/env python
"""Sandbox for exploring the new Payday algorithm.
"""
from __future__ import print_function, unicode_literals

from collections import defaultdict


# Classes

class _Thing(object):
    def __init__(self, name):
        self.name = name
        self.values = list()
    def __setitem__(self, k, v):
        self.values.append((k,v))
    def __repr__(self):
        return(self.name)
    def __str__(self):
        return '\n'.join(['{} {}'.format(repr(k),v) for k,v in self.values])

class Participant(_Thing):
    pass

class Team(_Thing):
    owner = None


# Universe

a, b, c, d, e = [Participant(x) for x in 'abcde']
A, B, C, D, E = [Team(x) for x in 'ABCDE']


# subscriptions

a[A] = 1
a[B] = 1
a[C] = 1
a[E] = 1

b

c[A] = 1
c[C] = 1
c[E] = 1

d[D] = 1

e[D] = 1


# payroll

A[b] = 1
A[c] = 1
A.owner = c

B.owner = c

C.owner = a

D.owner = d

E[c] = 1
E.owner = e


def payday(participants, teams):
    """Given a list of participants and a list of teams, return a list.

    The list we return contains instructions for funds transfer, both card
    captures (positive) and bank deposits (negative).

    """
    t_balances = defaultdict(int)
    p_balances = defaultdict(int)
    p_holding = defaultdict(int)

    for p in participants:
        for t, amount in p.values:
            t_balances[t] += amount
            p_holding[p] += amount

    for t in teams:
        for p, amount in t.values:
            t_balances[t] -= amount
            p_balances[p] += amount

    for t in teams:
        p_balances[t.owner] += t_balances[t]
        t_balances[t] -= t_balances[t]

    assert sum(t_balances.values()) == 0

    return [(p, p_balances[p] - p_holding[p]) for p in participants]


for participant, instruction in payday([a,b,c,d,e], [A,B,C,D,E]):
    print("{} {:2}".format(participant.name, instruction))
```

It *looks* like throwing a `Team` abstraction into payday makes the algorithm *much* nicer. It seems like it will enable us to easily settle the graph completely each cycle, *and* have no escrow! Who knew!? :dancer:

Any guesses what the above spits out? :-)

![payday](https://cloud.githubusercontent.com/assets/134455/7601658/7f2c5d9e-f8e7-11e4-9819-07389b5909f8.jpg)

(The "4" is wrong. It's zero, taking the sign into account [charge/deposit].)

**Comment by whit537** [comment]
@rohitpaulk Do you think there's any chance you'll be able to work on this before tomorrow's deadline? It'd be great to find a way to deal you in here. Maybe we can quickly stub out a `teams` table and `Team` class, and then you can work on the UI side while I work on the payday side?

**Comment by rohitpaulk** [comment]
@whit537 - Sure, sounds great. Let's cherry pick stuff out of #3400 and get a few more PRs out of the way.

**Comment by rohitpaulk** [comment]
Ah, looks like you're already on it :dancer:

**Comment by rohitpaulk** [comment]
Let's follow our common pattern - You start PRs, I'll build on them.

**Comment by whit537** [comment]
> Let's follow our common pattern - You start PRs, I'll build on them.

Perfect. I actually have to disappear for an hour or so to get kids out the door. I see you've merged #3410. Can you start on wiring up the UI for creating teams? I've started #3412 for that.

**Comment by whit537** [comment]
!m @rohitpaulk

**Comment by rohitpaulk** [comment]
On it. !m @whit537

**Comment by whit537** [comment]
@rohitpaulk Hopefully you're topped up with #3412 and #3413 for a bit. I'm going to dive in on #3414. It occurs to me that we don't even need the ability to modify tips before tomorrow. I expect to develop the new payments-related tables on #3414. Either in #3412 or in a separate PR we can implement migration from the old `tips` and `takes` tables to the new tables, and then after Payday 154 we can start bringing back the UI to work with payments and payroll.

**Comment by whit537** [comment]
Okay! Here's what I'm seeing:

 - @rohitpaulk is working on UI for creating, migrating, and approving teams.
 - @whit537 is rewriting payday (#3414)
 - @copiesofcopies 

@copiesofcopies In terms of timing, I'm going to be heads-down on [rewriting payday](https://github.com/gratipay/gratipay.com/pull/3414) for a while. We've got someone else [implementing](https://github.com/gratipay/gratipay.com/issues/3399#issuecomment-101620077) the team creation/migration/approval workflow. Those two plus the terms are the three big pieces we need in order to run payday tomorrow. The terms are the furthest along at this point. :-) Payday is the biggest challenge right now so I'm going to concentrate on that. My plan is to re-emerge later today to bring the terms in for a landing.

**Comment by whit537** [comment]
Gah, comment fail, sorry. :-/

**Comment by whit537** [comment]
We have until 3:30pm US/Pacific tomorrow to complete payday. We do want to get a couple people signed up first, though. I'm still going to spend some time right now working on payday (#3414), because it's risky and challenging and I need to work on it in the morning while I'm fresh, but we can finish coding that tomorrow (I do have a 10:00 [meeting](https://github.com/gratipay/gratipay.com/issues/3366#issuecomment-100329785) with Citizens). We really should land the terms (#3408) and the team sign-up process (#3412) today if we can, in order to give @dcht00 @fenryxo @silverhook @ehmatthes @mixolidia et al. a little time to get in the door for tomorrow's payday (#3415).

**Comment by whit537** [comment]
cc: @copiesofcopies

**Comment by whit537** [comment]
Okay! We're making progress on the terms (#3408). We've got a batch of 12 small proposals/questions to sort out, after which we'll hit two slightly larger questions, one about Open Work, and another about how Teams register. How are we doing with #3412, @rohitpaulk?

**Comment by whit537** [comment]
Oh, right. You already [told](https://github.com/gratipay/gratipay.com/pull/3412#issuecomment-101775486) me it was ready for review. :-)

!m @rohitpaulk

**Comment by whit537** [comment]
I've topped up #3412. Next step is to move #3414 far enough forward that we have our new payments schema; #3412 is blocked on that and #3408. Less important but easier to tackle is #3413.

@rohitpaulk: If you come back around before I'm able to make significant progress on #3412 or #3414, I see two things you could work on:

 - Try moving #3412 as far forward towards with migrations as you can. There's surely a bit of UI/testing work that could be done without actually having the schema present.
 - #3413

**Comment by whit537** [comment]
I just made a pass through support and replied to people who have been waiting to hear from us, asking them to wait a little longer ...

**Comment by whit537** [comment]
Alright, I realized we don't need to block sign-ups on migration, only on terms. Here's the current state of the sign-up form:

![screen shot 2015-05-13 at 8 49 44 pm](https://cloud.githubusercontent.com/assets/134455/7624129/b4845d3c-f9b1-11e4-96ef-b4a859cbf60f.png)
![screen shot 2015-05-13 at 8 49 56 pm](https://cloud.githubusercontent.com/assets/134455/7624132/b9b8f114-f9b1-11e4-82f9-26bfc471ef11.png)

It limits based on verified email and bank account or PayPal, and here's the success message (owner's email is displayed):

![screen shot 2015-05-13 at 8 52 15 pm](https://cloud.githubusercontent.com/assets/134455/7624153/076c47da-f9b2-11e4-9a78-221601760fe1.png)

**Comment by whit537** [comment]
Alright, the pressure is on #3408. Once we land that we can start signing up Teams! After that it's all about #3414 (with a little bit of #3413 and some migration work, which could fall under #3415).

**Comment by rohitpaulk** [comment]
#3413 is ready for review, and I've closed out support tickets that weren't related to the pivot.

**Comment by whit537** [comment]
Awwwweeeesssssoooommmmeee.

!m @rohitpaulk

**Comment by whit537** [comment]
Made it through support. Now to land #3408 and #3412 ...

**Comment by whit537** [comment]
#3408 landed. Almost there with #3412 ...

**Comment by whit537** [comment]
Okay! Applications for new teams are now open!

https://gratipay.com/new

**Comment by mattbk** [comment]
By "new teams" you mean anyone who wants to migrate, correct?  Not just new people to the service.

**Comment by fenryxo** [comment]
I got error "400 Bad Request: You must attach a bank account or PayPal to apply for a new team." although [~NuvolaPlayer](https://gratipay.com/~NuvolaPlayer) does have PayPal account attached. Should I create a new ticket for this?

**Comment by ehmatthes** [comment]
I applied for introtopython.org. I used my personal Gratipay account to do so; I'm not sure if that was the correct approach, but I only have one Gratipay account.

Also, I see a message that my credit card is working but out of date. I tried to update the credit card, but the information didn't seem to save. I'm not sure where this fits in with all the recent changes.

**Comment by whit537** [comment]
> By "new teams" you mean anyone who wants to migrate, correct? Not just new people to the service.

@mattbk Yes, that is correct.

> I got error "400 Bad Request: You must attach a bank account or PayPal to apply for a new team." although ~NuvolaPlayer does have PayPal account attached.

Bah, sorry. Looking into it ...

> I applied for introtopython.org.

Hooray! Introtopython.org is our first team! :dancer: !m @ehmatthes 

> I tried to update the credit card, but the information didn't seem to save.

Yeah, that's no doubt to regressions from migrating to Braintree. We are starting to store new cards there even though we're going to still be processing on Balanced today. You can make a ticket if you want, or look for a recent one that might be related ...

**Comment by whit537** [comment]
Back from Citizens (#3366). Todo:

 - [ ] fix paypal bug
 - [ ] get people signed up, specifically:
  - [ ] @fenryxo 
  - [ ] @silverhook
  - [ ] @mixolidia
  - [ ] Sudo Room
  - [ ] Sudo Mesh
 - [ ] rewrite payday (#3414)
 - [ ] migrate tips to subscriptions
 - [ ] run Gratipayday 154! :open_mouth: (#3415)
 - [ ] :cocktail: :beer: :poultry_leg: :love_hotel:

**Comment by rohitpaulk** [comment]
> fix paypal bug

On it.

**Comment by whit537** [comment]
!m @rohitpaulk

**Comment by whit537** [comment]
Alright, if you're on that, @rohitpaulk, then I'm going to rewrite payday (#3414). :rage1:

**Comment by rohitpaulk** [comment]
Cool, start sending commits as soon as you can, even if they're small - should help to have an extra pair of eyes on them :)

**Comment by whit537** [comment]
@rohitpaulk Good call.

**Comment by silverhook** [comment]
> I got error "400 Bad Request: You must attach a bank account or PayPal to apply for a new team." although ~NuvolaPlayer does have PayPal account attached. Should I create a new ticket for this?

I have the same issue.

**Comment by rohitpaulk** [comment]
@silverhook - We have a fix on #3423, will be deployed soon...

**Comment by whit537** [comment]
@rohitpaulk In general can you work on getting people signed up?

**Comment by rohitpaulk** [comment]
Yep, will do. You'll [have to deploy](https://github.com/gratipay/gratipay.com/pull/3423#issuecomment-102102260) #3423 though.

**Comment by whit537** [comment]
Deployed. Try again @fenryxo @silverhook? https://gratipay.com/new

**Comment by fenryxo** [comment]
@whit537 Works without any issues ;-)

**Comment by whit537** [comment]
Huzzah! :dancer: 

```
=> select name, owner from teams;
┌───────────────┬──────────────┐
│     name      │    owner     │
├───────────────┼──────────────┤
│ Gratipay      │ Gratipay     │
│ introtopython │ ehmatthes    │
│ Duo           │ mix          │
│ sudo room     │ sudoroom     │
│ Nuvola Player │ NuvolaPlayer │
└───────────────┴──────────────┘
(5 rows)
```

**Comment by whit537** [comment]
That is a good honkin' crew. Dang. :rocket: 

@silverhook Hopefully you can join us soon? Your trip is short. :dancer:

**Comment by silverhook** [comment]
× Looks like you've found a bug! Sorry for the inconvenience, we'll get it fixed ASAP!

**Comment by silverhook** [comment]
Ah, I got it – it doesn’t accept `/` in the name – I wanted to use `Cyberpipe/Kiberpipa hackerspace`

Submitted now :+1:

**Comment by whit537** [comment]
Yesssssssssss! :dancer:

**Comment by ehmatthes** [comment]
It's probably obvious, but I filled in the /new form in a very basic sense, with the intention of refining the copy as the new structure continues to take shape.

**Comment by ehmatthes** [comment]
PS !m everyone, it's been an exciting week to watch Gratipay development.

**Comment by rohitpaulk** [comment]
!m @ehmatthes :)

**Comment by whit537** [comment]
Okay, payday rewrite landed, proceeding with migration of `tips` to `subscriptions` ...

**Comment by whit537** [comment]
```
=> select name, owner, is_approved from teams;
┌───────────────┬──────────────┬─────────────┐
│     name      │    owner     │ is_approved │
├───────────────┼──────────────┼─────────────┤
│ Gratipay      │ Gratipay     │ t           │
│ introtopython │ ehmatthes    │ t           │
│ Duo           │ mix          │ t           │
│ sudo room     │ sudoroom     │ t           │
│ Nuvola Player │ NuvolaPlayer │ t           │
│ Cyberpipe     │ Kiberpipa    │ t           │
└───────────────┴──────────────┴─────────────┘
(6 rows)
```

**Comment by whit537** [comment]
/me deploys #3414 in order to pick up new schema ...

**Comment by whit537** [comment]
How's this look?

```sql
INSERT INTO subscriptions (ctime, subscriber, team, amount, is_funded)
(
    SELECT ctime
         , tipper
         , (SELECT slug FROM teams WHERE owner=tippee)
         , amount
         , is_funded
      FROM current_tips
      JOIN participants p
        ON p.username = tipper
     WHERE tippee IN (SELECT owner FROM teams WHERE team.is_approved)
       AND p.is_claimed IS NOT NULL
       AND p.is_suspicious IS NOT TRUE
       AND p.is_closed IS NOT TRUE
       AND amount > 0
);
```

/me testing on a backup ...

**Comment by rohitpaulk** [comment]
@whit537 - https://github.com/gratipay/gratipay.com/pull/3424. Basically the same thing, running the SQL directly would probably be faster than merging the PR and running the function on each of the teams though.

**Comment by whit537** [comment]
```
=# \i sql/migrate-tips-to-subscriptions.sql
INSERT 0 984
```

```sql
INSERT INTO subscriptions (ctime, subscriber, team, amount, is_funded)
(
    SELECT ct.ctime
         , ct.tipper
         , (SELECT slug FROM teams WHERE owner=tippee)
         , ct.amount
         , ct.is_funded
      FROM current_tips ct
      JOIN participants p
        ON p.username = tipper
     WHERE tippee IN (SELECT owner FROM teams WHERE is_approved)
       AND p.claimed_time IS NOT NULL
       AND p.is_suspicious IS NOT TRUE
       AND p.is_closed IS NOT TRUE
       AND amount > 0
);
```

**Comment by rohitpaulk** [comment]
`claimed_time`, not `is_claimed`..

**Comment by whit537** [comment]
Yeah, got it. Just ran it in production:

```
=> \i sql/migrate-tips-to-subscriptions.sql 
INSERT 0 984
```

O.O

**Comment by whit537** [comment]
Switching to #3415 ...

**Comment by whit537** [comment]
I made another pass through support to keep people topped up. We've got 14 open tickets there that we'll need to follow up on as things settle down. Next steps are to complete and clean up after payday (#3415), and make a blog post announcing the change (#3419). After that let's remaining tickets on the [Pivot](https://github.com/gratipay/gratipay.com/milestones/Pivot) milestone. This ticket'll be the last out the door.

**Comment by whit537** [comment]
Okay! Big day. :100: 

Today we need to announce Gratipay 2.0 (#3419), so that we can get more people signed up in time for payday 155 tomorrow. There are a few things we should clean up before we publish that announcement. #3432 is a good example. There may be others as I think through this here ...

You around today, @rohitpaulk?

**Comment by rohitpaulk** [comment]
Yep, I'm here. Should I be focusing on Braintree or the Pivot?

**Comment by rohitpaulk** [comment]
Most of the cached amounts work has been done in #3449, should just be a matter of picking it out of the mess.

**Comment by whit537** [comment]
@rohitpaulk Huzzah! Pivot today, Braintree tomorrow. We need to get this announcement out the door ASAP so people can start signing up again before tomorrow. We're going to have to bump payroll and stats/caching past tomorrow, I'm afraid. :-/

Blockers for #3419 are at https://github.com/gratipay/gratipay.com/issues/3419#issuecomment-103870467.

**Comment by rohitpaulk** [comment]
> We're going to have to bump payroll and stats/caching past tomorrow

Payroll is done in #3450, ready for review. Stats/caching will take a while, yh. 

@whit537 - I'll work on #2387.

**Comment by whit537** [comment]
> I'll work on #2387.

@rohitpaulk Perfect, thank you. I am finishing up some revisions to #3419 and then I'm going to handle https://github.com/gratipay/gratipay.com/issues/3366#issuecomment-103895393 on my way to the coffee shop. I will reappear in a couple hours with the goal of landing #3419 by suppertime.

**Comment by whit537** [comment]
Announcement is out the door! We have launched [Gratipay 2.0](https://medium.com/gratipay-blog/gratipay-2-0-2453d3c53077)! :dancer: 

!m *

**Comment by whit537** [comment]
We pivoted (#180) to [Gratipay 2.0](https://medium.com/gratipay-blog/gratipay-2-0-2453d3c53077). Now, we need to turn the :ship: a couple degrees more.

- Gratipay 2.0: Payments and payroll for open work.
- **Gratipay 2.1**: Payments and payroll for open *companies*.

Why? As [before](https://github.com/gratipay/inside.gratipay.com/issues/180#issuecomment-100497390), we're being driven negatively by legal concerns, but also positively by business vision.

Gratipay 2.0 moved us clearly into [the safe zone](https://github.com/gratipay/inside.gratipay.com/issues/201#issuecomment-111157623) regarding direct BSA regulation, but we still have work to do to [comply with AML law](https://github.com/gratipay/inside.gratipay.com/issues/119) to the satisfaction of our prospective processing partners (currently #3491; see also: #3366 #417 #481). Separately, in trying to [bring back payroll](https://github.com/gratipay/gratipay.com/issues/3433) we've started wrestling with our story [relative to employment law](https://github.com/gratipay/inside.gratipay.com/issues/242). AML and employment law are the two negative constraints driving Gratipay 2.1.

The principle of business vision that is driving Gratipay 2.1 is that we want **economic engines**, because our [mission](http://inside.gratipay.com/big-picture/mission) is to cultivate **an economy**. We want *companies*. We don't just want "teams" that don't make any money. We want **companies** that make money *and* offer open work. We want **open companies**.

**Comment by whit537** [comment]
Here are the changes involved in Gratipay 2.1:

 - Stop talking about teams and start talking about open companies.
 - Attach a withdrawal route directly to the company.
 - Stop talking about company *owners* and start talking about company *managers*.
 - Allow for more than one manager.
 - Allow for setting a withholding amount on individual takes.
 - Takes now represent contractors, employees, and/or owners.
 - Allow for managers to set takes rather than members always setting their own takes.
 - Collect and verify company identity for all companies.
 - Collect and verify personal identity for all members.

**Comment by tshepang** [comment]
Can't we keep things simple and only do teams?

**Comment by whit537** [comment]
@tshepang By "teams" I take it you are referring to the friction-free revenue sharing that we had under Gratipay 1.0. Unfortunately, that has been an over-simplification, lulling our users into a false sense of security. The best explanation we've had yet for the legal relationship between teams and members under Gratipay 1.0 is that the team [isn't actually a party to the transaction](https://github.com/gratipay/gratipay.com/issues/2569#issuecomment-86491103):

> [T]he members don't receive money from Gratipay LLC, they receive money from anonymous tippers.

And again at https://github.com/gratipay/inside.gratipay.com/issues/211#issuecomment-109513994: 

> I was under the assumption that Gratipay would function to make the distributions directly to the different members and each of us would be obligated to self-report our own income on our Schedule C

As I responded [there](https://github.com/gratipay/inside.gratipay.com/issues/211#issuecomment-109518669), I don't think this line of thinking bears examination, because the flow of funds still has to be accounted for, and the only way I see of accounting for the flow of funds under the 1.0 model is too much of a stretch to be tenable:

> Let's say Company A uses jshttp and pays $100, and that gets split between 10 developers on the jshttp team. Are those 10 each independent contractors of Company A? Does Company A get ten invoices, one from each of them, for their portion of the $100 payment? It seems less strained to me to say that jshttp is itself a company (a legal entity), that receives $100 in income from Company A, and pays the 10 developers either as contractors, employees, or legal owners of the jshttp legal entity.

**Comment by fenryxo** [comment]
> Gratipay 2.1: Payments and payroll for open companies.

Is it necessary to have a company or is a sole proprietorship as a legal entity (+ contractors) enough?

**Comment by whit537** [comment]
> [T]he only way I see of accounting for the flow of funds under the 1.0 model is too much of a stretch to be tenable[.]

Or, at least, to be safe. The model where jshttp is a company receiving income and distributing payroll—that model is well-established and understood, even if it is complex. If we say that Company A is independently contracting with 10 individuals, then we're going against the grain of the global economy and introducing a new model. I don't think that's where we want to spend our energy. Play it out: we get big enough for someone to care whether our model is legal, and the way questions of the form "Is this legal?" get answered is through litigation. Establishing new law is not our modus operandi. Our MO is finding ways to build our abstractions atop existing law.

**Comment by whit537** [comment]
> Is it necessary to have a company or is a sole proprietorship as a legal entity (+ contractors) enough?

@fenryxo At this point I believe a sole proprietorship will be fine. I would love to find a way to provide a "company as a service," but for the foreseeable future Gratipay is going to have to offload nearly all of the responsibility for accounting for contractors, employees, and/or owners to you, the team/company customer.

**Comment by whit537** [comment]
@fenryxo Have you used contractors in the past? Is it a process you're familiar/comfortable with? Was that for Nuvola or was that in other contexts?

**Comment by fenryxo** [comment]
@whit537. No, I haven't used contractors yet as the Nuvola Player funding haven't reached the threshold for revenue sharing. However, I've talked to my accountant/tax advisor about this and it would be feasible to have contractors once project funding gets higher. On the other hand, it makes no sense to set up a (limited liability) company until really big money are involved.

**Comment by whit537** [comment]
> However, I've talked to my accountant/tax advisor about this and it would be feasible to have contractors once project funding gets higher.

Awesome. :-)

> On the other hand, it makes no sense to set up a (limited liability) company until really big money are involved.

Some day! :-)

**Comment by silverhook** [comment]
> > Gratipay 2.1: Payments and payroll for open companies.

> Is it necessary to have a company or is a sole proprietorship as a legal entity (+ contractors) enough?

Will an NGO/NPO be enough? Or will hackerspaces fall through this filter?

**Comment by whit537** [comment]
> Will an NGO/NPO be enough?

Definitely enough.

> Or will hackerspaces fall through this filter?

I hope not! :-)

Echoing @ehmatthes comments at https://github.com/gratipay/inside.gratipay.com/issues/251#issuecomment-112550580, I'm using the term "open company" loosely. A non-profit can definitely be "an open company in gratipay's eyes."

**Comment by whit537** [comment]
Alright, let's bring this in for a landing. We have 24 tickets still open on the [Pivot](https://github.com/gratipay/gratipay.com/milestones/Pivot) milestone, and this is the main ticket.

**Comment by whit537** [comment]
I've been looking at site copy (https://github.com/gratipay/gratipay.com/issues/3398) today, specifically [the About pages](https://gratipay.com/about/). I guess I was feeling overwhelmed and that seemed like an easy place to start that will certainly provide value when we drive traffic for #3539. It also solidifies the work we did to [standardize nomenclature](https://github.com/gratipay/inside.gratipay.com/issues/117), and queues us up to revise the Team application in https://github.com/gratipay/gratipay.com/issues/3677.

**Comment by webmaven** [comment]
@whit537:
> Echoing @ehmatthes comments at gratipay/inside.gratipay.com#251 (comment), I'm using the term "open company" loosely. A non-profit can definitely be "an open company in gratipay's eyes."

So, perhaps 'open organizations' could be an improvement? This would be inclusive of companies, non-profits, co-ops, and (potentially) eventually other more exotic arrangements (such as [decentralized autonomous organizations](https://en.wikipedia.org/wiki/Decentralized_autonomous_organization)). We can link the term to the [glossary](/gratipay/inside.gratipay.com/issues/243), although a tooltip would perhaps be even better.

**Comment by webmaven** [comment]
It would also likely be helpful if the [gratipay](/gratipay/gratipay.com) and [inside.gratipay.com](/gratipay/inside.gratipay.com) repos were to have 2.1 milestones.

**Comment by tshepang** [comment]
:+1: @webmaven

**Comment by whit537** [comment]
+1 for "open organizations"

**Comment by mitar** [comment]
Hm, but how does PayPal handle the fact that they keep money in their accounts and not immediately transfer the money? Are they incorporated as a bank?

**Comment by rohitpaulk** [comment]
Paypal is registered as an MSB (money services business), yes. https://www.paypal.com/webapps/mpp/licenses

**Comment by whit537** [comment]
Still, PayPal's licensing applies to the *transfer* of money, not to the storage of it. As best I can tell, they avoid regulation as a bank in the U.S. largely by asserting that they're not a bank. :-)

We've talked about this elsewhere, I believe. I'm not sure this is the best place for this convo.

**Comment by Franklewis1978** [comment]
#942 Rosario 600$ @ car

**Comment by whit537** [comment]
Hmmm ... let's call that spam. Deleted!

**Comment by whit537** [comment]
> So, perhaps 'open organizations' could be an improvement?

You anticipated Jim Whitehurst, @webmaven. ;-)

https://www.redhat.com/en/explore/the-open-organization-book

cf. https://github.com/gratipay/inside.gratipay.com/issues/672

**Comment by whit537** [comment]
> We have 24 tickets still open on the [Pivot](https://github.com/gratipay/gratipay.com/milestones/Pivot) milestone, and this is the main ticket.

We're now down to eight!

**Comment by whit537** [comment]
Well, we did pivot another degree, but towards [projects](https://gratipay.news/from-teams-to-projects-45c46718507b) instead of [open companies](https://github.com/gratipay/gratipay.com/issues/3399#issuecomment-112110512):

> We still believe in open work, but our new goal is to onboard a million projects with 10% offering open work, instead of ten thousand with 100% offering open work.

Now the Pivot is down to one last ticket: this one. Time to close it out!

Congratulations, everyone! We have successfully turned the ship! 💃 🌻
